### PR TITLE
feat(frontend): Badge component and script page

### PR DESCRIPTION
**Pull request by adam-kov**
Created a common Badge component, fixed the inconsistensies of the tooltip text and implemented a new design for the script detail page.
**Button functionalities on script page are unchanged in this PR.**

**Desktop:**
desktop

**Mobile:**
mobile

### CODE REVIEW

**Comment by adam-kov** [comment]
I have read the CLA Document and I hereby sign the CLA

File: frontend/src/lib/components/common/badge/Badge.svelte [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+	import { classNames } from '$lib/utils'
+	import { CloseButton } from 'flowbite-svelte'
```
Comment (fatonramadani) [inline frontend/src/lib/components/common/badge/Badge.svelte:3]: We might want to stop using `flowbite-svelte' in the future.

**Comment by rubenfiszel** [comment]
Nice work @adam-kov 

@fatonramadani has taken the habits of putting a screenshot in the PR description for UI/UX changes which I quite like. Wdyt about doing that ? It's good for the visibility of your work. E.g: https://github.com/windmill-labs/windmill/pull/612

**Comment by adam-kov** [comment]
Yes, I forgot to take screenshots, my bad.

**Desktop:**
desktop

**Mobile:**
mobile

File: frontend/src/lib/components/common/badge/model.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import type { IconProps } from 'svelte-awesome/components/Icon.svelte'
+
+type BaseColor = 'blue' | 'gray' | 'red' | 'green' | 'yellow' | 'indigo' | 'purple' | 'pink'
+export const ColorModifier = 'dark-'
```
Comment (fatonramadani) [inline frontend/src/lib/components/common/badge/model.ts:4]: Dark mode should not be set this way:  
Each color (blue, gray, ...) should define its dark colors using the tailwind `dark:` variant.
[Tailwind Dark mode documentation](https://tailwindcss.com/docs/dark-mode)

File: frontend/src/lib/components/SharedBadge.svelte [hunk 3]
```diff
@@ -49,14 +48,9 @@
...
-	<span class="mr-1 align-center">
-		{#if kind == 'read'}
-			<Badge tooltip={reason}>
-				<Icon data={faPeopleGroup} scale={0.7} />
-				read</Badge
-			>
-		{:else if kind == 'write'}
-			<Badge tooltip={reason}><Icon data={faPeopleGroup} scale={0.7} /></Badge>
+	<Badge icon={{ data: faPeopleGroup }} color="blue">
+		{#if kind === 'read'}
```
Comment (fatonramadani) [inline frontend/src/lib/components/SharedBadge.svelte]: `if` statement should be around the badge.

File: frontend/src/lib/components/common/badge/Badge.svelte [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	export let icon: BadgeIconProps | undefined = undefined
+
+	let defaulIconProps: BadgeIconProps = {
+		data: undefined,
+		position: 'left',
+		scale: 0.7
+	}
+	let hidden = false
+	const colors: Record<BadgeColor, string> = {
+		gray: 'bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300',
```
Comment (fatonramadani) [inline frontend/src/lib/components/common/badge/Badge.svelte:24]: As a rule a thumb, please add as little code as necessary and we'll add more colors if we need them later.

File: frontend/src/lib/components/common/badge/model.ts (hunk 2 above)
Comment (adam-kov) [reply frontend/src/lib/components/common/badge/model.ts:4]: The intention here wasn't about creating dark mode variants, but to provide a set of darker shades for the colors.

**Review by fatonramadani — APPROVED** [review]

**Pull request merged**